pycbc/psd/variation.py

```diff
@@ -160,3 +165,214 @@ def find_trigger_value(psd_var, idx, start, sample_rate):
     ind -= 1
     vals = psd_var[ind]
     return vals
+
+
+def mean_square(data, delta_t, short_stride, stride):
+    """ Calculate mean square of given time series once per stride
+
+    First of all this function calculate the mean square of given time
+    series once per short_stride. This is used to remove outliers due
+    to short glitches. Then, every seconds it computes the mean square
+    of the smoothed time series, within the stride.
+
+    Parameters
+    ----------
+    data : TimeSeries
+    delta_t : float
+        Duration of the time series
+    short_stride : float
+        Stride duration for outlier removal
+    stride ; float
+        Stride duration
+
+    Returns
+    -------
+    m_s: List
+        Mean square of given time series
+    """
+
+    srate = int(data.sample_rate)
+
+    # Calculate mean square of data once per short stride and remove
+    # the outliers
+    data = numpy.array(data)
+    short_ms = numpy.mean(data.reshape(-1, int(srate * short_stride)) ** 2,
+                          axis=1)
+    ave = 0.5 * (short_ms[2:] + short_ms[:-2])
+    outliers = short_ms[1:-1] > (2 * ave)
```

Comment:
add comment here explaining why the 2 * ave definition of 'outlier' is chosen - also maybe prefer `2.` ie explicitly a float.